File: frontend/src/features/submission/model/SubmissionJudgeCompletion.ts
import type { SubmissionStatus } from '@/features/submission/model/SubmissionStatus'
import type { SubmissionVerdict } from '@/features/submission/model/SubmissionVerdict'

export type SubmissionJudgeCompletion = {
  status: SubmissionStatus
  verdict: SubmissionVerdict | null
  judgeMessage: string | null
}

File: backend/src/main/scala/domains/submission/model/SubmissionJudgeCompletion.scala
package domains.submission.model

import judgeprotocol.model.JudgeResult

final case class SubmissionJudgeCompletion(
  status: SubmissionStatus,
  verdict: Option[SubmissionVerdict],
  judgeMessage: Option[String],
  timeUsedMs: Option[Long],
  memoryUsedKb: Option[Long],
  score: Option[BigDecimal],
  judgeResult: Option[JudgeResult]
)
